test/sasdataloader/test/utest_cansas.py

```diff
@@ -104,20 +101,18 @@ def _check_data(self, data):
         self.assertTrue(data.process[0].name == "Mantid generated CanSAS1D XML")
         self.assertTrue(data.meta_data["xmlpreprocess"] is not None)
 
-
     def _check_data_1_1(self, data):
         spectrum = data.trans_spectrum[0]
         self.assertTrue(len(spectrum.wavelength) == 138)
 
-
     def test_cansas_xml(self):
-        filename = "isis_1_1_write_test.xml"
         xmlreader = XMLreader(self.isis_1_1, self.schema_1_1)
         valid = xmlreader.validate_xml()
         xmlreader.set_processing_instructions()
         self.assertTrue(valid)
         fo = open(self.isis_1_1)
-        str = fo.read()
+        fo.read()
+        fo.close()
```

Comment:
Should be:

    with open(self.isis_1_1) as fo:
        text = fo.read()

Neither matter in this case because it is a test, but for long running programs it is better to use the with ... construct so that the file is closed immediately even in the event of an error in reading.  Also, replacing builtins like 'str' is usually a bad idea (though min() and max() are sitting on some nice name realestate).